### PR TITLE
Add Hacktoberfest guide

### PR DESCRIPTION
**Pull request by yanthomasdev**
This PR adds the Hacktoberfest guide to be shared as our official entry point to the event. Need to confirm which things to highlight and confirm the other participating repos.

### CODE REVIEW

File: astro.config.mjs [hunk 1]
```diff
@@ -82,6 +82,7 @@ export default defineConfig({
 						{ label: 'Adding a new Astro + X Guide', link: '/guides/new-third-party-stub/' },
 						{ label: 'Contributing a Recipe', link: '/guides/recipe-writing/' },
 						{ label: 'Translating Astro docs', link: '/guides/i18n/' },	
+						{ label: 'Contributing on Hacktoberfest', link: '/guides/hacktoberfest/' },
```
Comment (sarah11918) [inline astro.config.mjs]:
```suggestion
						{ label: 'Contributing for Hacktoberfest', link: '/guides/hacktoberfest/' },
```

File: src/content/docs/guides/hacktoberfest.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+title: Hacktoberfest
+description: Learn how to contribute to Astro during Hacktoberfest.
+---
+
+[Hacktoberfest](https://hacktoberfest.com/about/) is an annual event during the month of October to encourage open-source contributions. If you've ever wanted to contribute to Astro, this is your opportunity. From beginner to experienced, from developer to low/no-code contributor, you're welcome!
+
+:::note
+Since Hacktoberfest happens once a year during October, assume the information on this page to be only valid and up-to-date during October of each year.
+
+Things such as participating repos, rules, instructions, rewards, and so on may vary from year to year.
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
Participating Astro repositories, rules, instructions, and rewards may vary from year to year. Please consult this guide for up-to-date information every October!
```

File: astro.config.mjs (hunk 1 above)
Comment (sarah11918) [inline astro.config.mjs]: My first thought is this is pretty buried on the page, so let's have a banner (OK to add with this PR and have shown as soon as it merges.)

File: src/content/docs/guides/hacktoberfest.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+To take part in Hacktoberfest, here's all you need to do:
+
+1. Register for the event on [Hacktoberfest's official website](https://hacktoberfest.com/).
+2. Make any [valid contributions](#contributions) following [Hacktoberfest's rules](https://hacktoberfest.com/participation/#pr-mr-details) during October.
+3. Let us know you're participating by requesting your PRs to be labeled as `hacktoberfest-accepted`.
+
+## Primer on open-source
+
+Especially for first-time contributors, open-source can be quite frightening -- but don't worry, we're here to help you succeed! Here are a few reliable resources you can read to get up to speed on the matter:
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
Especially for first-time contributors, open-source can be quite frightening -- but don't worry, we're here to help you succeed! 

In addition to [our own resources for first-time contributors](/first-time/1-about/), here are a few reliable resources to help you prepare for your first Hacktoberfest contributions:
```

File: src/content/docs/guides/hacktoberfest.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+## Join the fest
+
+To take part in Hacktoberfest, here's all you need to do:
+
+1. Register for the event on [Hacktoberfest's official website](https://hacktoberfest.com/).
+2. Make any [valid contributions](#contributions) following [Hacktoberfest's rules](https://hacktoberfest.com/participation/#pr-mr-details) during October.
+3. Let us know you're participating by requesting your PRs to be labeled as `hacktoberfest-accepted`.
+
+## Primer on open-source
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
## How to contribute to open-source
```

File: src/content/docs/guides/hacktoberfest.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Especially for first-time contributors, open-source can be quite frightening -- but don't worry, we're here to help you succeed! Here are a few reliable resources you can read to get up to speed on the matter:
+
+- [What is open-source](https://www.digitalocean.com/community/tutorials/what-is-open-source)
+- [How to contribute to open-source](https://opensource.guide/how-to-contribute/)
+- [How to use Git](https://www.digitalocean.com/community/cheatsheets/how-to-use-git-a-reference-guide)
+- [Getting started with GitHub Desktop](https://docs.github.com/en/desktop/overview/getting-started-with-github-desktop)
+
+## Contributions
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
## Allowed contributions
```
Not sure what exactly to put here, but allowable, accepted, etc. this heading should state that it is telling what is or isn't going to be accepted

File: src/content/docs/guides/hacktoberfest.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+Any contribution to [participating repositories](#participating-repositories) will be considered and count towards Hacktoberfest. Our only exceptions are contributions that are either spam attempts or that have no quality and effort.
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
Astro welcomes contributions of all sizes from contributors of any experience level to this year's [participating repositories](#participating-repositories). **To be considered for Hacktoberfest credit, the contribution must improve the quality of the repository with a helpful addition, improvement, or fix.**

Small fixes such as correcting typos are welcome and appreciated! Larger contributions such as translating a page of documentation or submitting a fix to an existing issue may also count towards Hacktoberfest totals if the PR can be reviewed and merged during the month of October.

We will be closing PRs that do not meet our quality standards. Spam submissions will not be tolerated, and may result in being banned from the repo.
```

File: src/content/docs/guides/hacktoberfest.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+:::caution
+AI contributions will not be accepted. Hacktoberfest exists to incentivize human contribution to open-source projects, and therefore, we cannot accept contributions that have been made by automated tools.
+
+This does not apply to AI assistance tools, such as GitHub Copilot or Cursor, that do not take over the core work of contributing from you.
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
**AI contributions will not be accepted.** Hacktoberfest exists to motivate and reward human contribution to open-source projects, and therefore, we cannot accept contributions that have been made by automated tools.

AI assistance tools, such as GitHub Copilot or Cursor, may be used to help you draft your contribution, but must be checked and edited for accuracy. If your submission contains obviously AI-generated errors, your PR will be rejected and you may be banned from further Hacktoberfest participation at our discretion.
```

File: src/content/docs/guides/hacktoberfest.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+Here are some of the main ways you can contribute to it:
+
+- Reading the current or [5.0 beta docs](https://5-0-0-beta.docs.astro.build/) and fixing any typos, grammar issues, or missing information.
+- Helping [translate the documentation](http://contribute.docs.astro.build/guides/i18n/) in your language.
+- Tackling [open "good first issues"](https://github.com/withastro/docs/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) in the repository.
+
+:::note[TODO]
+Adding information on the other participating repositories, I need to confirm.
+:::
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
:::note[Other repositories]
Other Astro repositories may agree to add the `hacktoberfest-accepted` tag on a case-by-case basis.
:::
```

File: src/content/docs/guides/hacktoberfest.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+## Rewards
+
+All participating contributors will get thematic [Hacktoberfest Astro Badges](https://astro.badg.es/achievements/hacktoberfest-merges/bronze/) that "level up" as you contribute. You can showcase your Astro badges in a daily updated profile card, like this one:
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
Astro allows you to earn [Hacktoberfest achievements on your Astro Badge](https://astro.badg.es/achievements/hacktoberfest-merges/bronze/) that "level up" as you contribute. Every contributor to Astro automatically has a profile card generated that tracks several achievements, like this one: 
```

File: src/content/docs/guides/hacktoberfest.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+Adding information on the other participating repositories, I need to confirm.
+:::
+
+## Rewards
+
+All participating contributors will get thematic [Hacktoberfest Astro Badges](https://astro.badg.es/achievements/hacktoberfest-merges/bronze/) that "level up" as you contribute. You can showcase your Astro badges in a daily updated profile card, like this one: 
+
+[![@jsparkdev Astro contributions](https://astro.badg.es/v2/contributor/jsparkdev.svg)](https://astro.badg.es/contributor/jsparkdev/)
+
+
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]: [Enter your GitHub username and find your own contributor badge!](https://astro.badg.es/contributors/)

this wouldn't let me add somehow as a suggestion

**Review by sarah11918 — COMMENTED** [review]
Amazing Yan! :partying_face: Left some thoughts for us to consider!

File: src/content/docs/guides/hacktoberfest.md (hunk 10 above)
Comment (yanthomasdev) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
[Enter your GitHub username and find your own contributor badge!](https://astro.badg.es/contributors/)
```

File: astro.config.mjs (hunk 1 above)
Comment (yanthomasdev) [reply astro.config.mjs]: I assume you mean a banner on the homepage? I'll add one and then you can reword/change places if you want!

File: src/content/docs/guides/hacktoberfest.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+Participating Astro repositories, rules, instructions, and rewards may vary from year to year. Please consult this guide for up-to-date information every October!
+:::
+
+## Join the fest
+
+To take part in Hacktoberfest, here's all you need to do:
+
+1. Register for the event on [Hacktoberfest's official website](https://hacktoberfest.com/).
+2. Make any [valid contributions](#allowed-contributions) following [Hacktoberfest's rules](https://hacktoberfest.com/participation/#pr-mr-details) during October.
+3. Let us know you're participating by requesting your PRs to be labeled as `hacktoberfest-accepted`.
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
3. Let us know you're participating by requesting the PRs label `hacktoberfest-accepted`.
```

**Review by sarah11918 — APPROVED** [review]
Thank you so much for putting this excellent guide together Yan! I know some of the other repos have responded that they are participating, so we can add them now or as we go, but this is ready for publication i think!

File: src/content/docs/guides/hacktoberfest.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+**AI contributions will not be accepted.** Hacktoberfest exists to motivate and reward human contributions to open-source projects, and therefore, we cannot accept contributions that have been made by automated tools.
+
+AI assistance tools, such as GitHub Copilot or Cursor, may be used to help you draft your contribution but must be checked and edited for accuracy. If your submission contains obviously AI-generated errors, your PR will be rejected and you may be banned from further Hacktoberfest participation at our discretion.
+:::
+
+## Participating repositories
+
+### Astro Docs
+
+The [`withastro/docs` repository](https://github.com/withastro/docs) contains the complete source code and content for the [Astro documentation](https://docs.astro.build/) website in all the available languages.
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
[`withastro/docs`](https://github.com/withastro/docs) contains the complete source code and content for the [Astro documentation](https://docs.astro.build/) website in all the available languages.
```
This formatting looked a little ugly/overwhelming visually, so let's optimize for people scanning and looking for repos!

File: src/content/docs/guides/hacktoberfest.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+Here are some of the main ways you can contribute to it:
+
+- Reading the current or [5.0 beta docs](https://5-0-0-beta.docs.astro.build/) and fixing any typos, grammar issues, or missing information.
+- Helping [translate the documentation](http://contribute.docs.astro.build/guides/i18n/) in your language.
+- Tackling [open "good first issues"](https://github.com/withastro/docs/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) in the repository.
+
+:::note[Other repositories]
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md:71]:
```suggestion
### Astro Adapters

[/withastro/adapters](https://github.com/withastro/adapters) is the repository for Astro's official adapter integrations: Node.js, Netlify, Vercel, and Cloudflare.

### Houston Discord Bot

[/withastro/houston-discord](https://github.com/withastro/houston-discord) is Astro's own Houston, a bot made to make the life of everyone within the Astro community easier.

:::note[Other repositories]
```

File: src/content/docs/guides/hacktoberfest.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+### Astro Docs
+
+[`withastro/docs`](https://github.com/withastro/docs) contains the complete source code and content for the [Astro documentation](https://docs.astro.build/) website in all the available languages.
+
+Here are some of the main ways you can contribute to it:
+
+- Reading the current or [5.0 beta docs](https://5-0-0-beta.docs.astro.build/) and fixing any typos, grammar issues, or missing information.
+- Helping [translate the documentation](http://contribute.docs.astro.build/guides/i18n/) in your language.
+- Tackling [open "good first issues"](https://github.com/withastro/docs/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) in the repository.
```
Comment (sarah11918) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
- Tackling open [ "good first issues"](https://github.com/withastro/docs/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) (or if you're more familiar with Astro, ["help wanted"](https://github.com/withastro/docs/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) issues) in the repository.
```

Comment (sarah11918) [reply src/content/docs/guides/hacktoberfest.md]: i'll let you edit this Yan, just wanted to also point out somehow that there is a separate help wanted for not necessarily :baby_bottle: issues!

File: src/content/docs/guides/hacktoberfest.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+
+### Astro Adapters
+
+[/withastro/adapters](https://github.com/withastro/adapters) is the repository for Astro's official adapter integrations: Node.js, Netlify, Vercel, and Cloudflare.
```
Comment (HiDeoo) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
[`withastro/adapters`](https://github.com/withastro/adapters) is the repository for Astro's official adapter integrations: Node.js, Netlify, Vercel, and Cloudflare.
```

To match the Astro Docs styling just above.

File: src/content/docs/guides/hacktoberfest.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+
+### Houston Discord Bot
+
+[/withastro/houston-discord](https://github.com/withastro/houston-discord) is Astro's own Houston, a bot made to make the life of everyone within the Astro community easier.
```
Comment (HiDeoo) [inline src/content/docs/guides/hacktoberfest.md]:
```suggestion
[`withastro/houston-discord`](https://github.com/withastro/houston-discord) is Astro's own Houston, a bot made to make the life of everyone within the Astro community easier.
```

File: src/content/docs/guides/hacktoberfest.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+
+:::note[Other repositories]
```
Comment (HiDeoo) [inline src/content/docs/guides/hacktoberfest.md:71]:
```suggestion
### Starlight

[`withastro/starlight`](https://github.com/withastro/docs) is a full-featured documentation theme built on top of the Astro framework.

You can contribute to the Starlight theme by improving the [documentation](https://starlight.astro.build/) by fixing typos, grammar issues, missing information or by helping [translate](https://github.com/withastro/starlight/blob/main/CONTRIBUTING.md#translations) Starlight’s UI or documentation in your language.

:::note[Other repositories]
```

**Review by HiDeoo — COMMENTED** [review]
Just realized we only talked about Starlight and Hacktoberfest on Discord and I totally forgot to suggest it here, sorry. Feel free to tweak the wording if needed.

**Pull request merged**